sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -259,6 +262,7 @@ case class StaticInvoke(
 
   override def nullable: Boolean = needNullCheck || returnNullable
   override def children: Seq[Expression] = arguments
+  override lazy val deterministic: Boolean = isDeterministic && arguments.forall(_.deterministic)
```

Comment:
nit: seems we can move this to `InvokeLike`

Reply:
I thought about this too, but then we'd have to move `isDeterministic` there too and then override it in `StaticInvoke` and `Invoke`, so doesn't seem we can save much. Plus I think deterministic property is not useful for `NewInstance`.